src/scripts/clipperUI/clipperStateUtilities.ts

```diff
@@ -1,13 +1,81 @@
-		return (state.userResult && state.userResult.status && state.userResult.data && !!state.userResult.data.user);
+	export function isUserLoggedIn(state: ClipperState, refreshUserInfo?: boolean): boolean {
+		if (state.userResult && state.userResult.status && state.userResult.data && !!state.userResult.data.user) {
+			return true;
+		} else if (!!refreshUserInfo) {
+			/**
+			 * Refresh the user info if the user is logged in and the user info is not available.
+			 * This could have possibly happened due to inactivity of the service worker.
+			 */
+			sendToOffscreenDocument(OffscreenMessageTypes.getFromLocalStorage, {
+				key: ClipperStorageKeys.isUserLoggedIn
+			}).then((isUserLoggedIn) => {
+				if (isUserLoggedIn === "true") {
+					let userInfoUpdateCb = (updatedUser: UserInfo) => {
+						if (updatedUser) {
+							let userInfoUpdatedEvent = new Log.Event.BaseEvent(Log.Event.Label.UserInfoUpdated);
+							userInfoUpdatedEvent.setCustomProperty(Log.PropertyName.Custom.UserUpdateReason, UpdateReason[updatedUser.updateReason]);
+							userInfoUpdatedEvent.setCustomProperty(Log.PropertyName.Custom.LastUpdated, new Date(updatedUser.lastUpdated).toUTCString());
+							Clipper.logger.logEvent(userInfoUpdatedEvent);
+						}
+
+						if (updatedUser && updatedUser.user) {
+							let timeStampedData: TimeStampedData = {
+								data: updatedUser.user,
+								lastUpdated: updatedUser.lastUpdated
+							};
+
+							// The user SV should never be set with expired user information
+							let tokenHasExpiredForLoggedInUser = CachedHttp.valueHasExpired(timeStampedData, (updatedUser.user.accessTokenExpiration * 1000) - 180000);
+							if (tokenHasExpiredForLoggedInUser) {
+								Clipper.logger.logFailure(Log.Failure.Label.UserSetWithInvalidExpiredData, Log.Failure.Type.Unexpected);
+							}
+
+							state.setState({ userResult: { status: Status.Succeeded, data: updatedUser } });
+							Clipper.logger.setContextProperty(Log.Context.Custom.AuthType, updatedUser.user.authType);
+							Clipper.logger.setContextProperty(Log.Context.Custom.UserInfoId, updatedUser.user.cid);
+						} else {
+							state.setState({ userResult: { status: Status.Failed, data: updatedUser } });
+						}
+					};
+
+					state.setState({ userResult: { status: Status.InProgress } });
+					Clipper.getExtensionCommunicator().callRemoteFunction(Constants.FunctionKeys.getInitialUser, {
+						callback: (freshInitialUser: UserInfo) => {
+							if (freshInitialUser && freshInitialUser.user) {
+								Clipper.logger.logUserFunnel(Log.Funnel.Label.AuthAlreadySignedIn);
+							} else if (!freshInitialUser) {
+								userInfoUpdateCb(freshInitialUser);
+							}
+							Clipper.getExtensionCommunicator().subscribeAcrossCommunicator(new SmartValue<UserInfo>(), Constants.SmartValueKeys.user, (updatedUser: UserInfo) => {
+								userInfoUpdateCb(updatedUser);
+							});
+						}
+					});
+				}
+			});
+			/**
```

Comment:
Didn't face any issue while testing but as per this comment - is there a chance that for a logged in user we will momentarily see the sign in panel and then once the offscreen task is completed we'll see the logged in state?

Reply:
I have tested sign in and sign out multiple times with the changes in this PR, and each time I was able to see the correct panel.